src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -414,6 +419,70 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       particleGun->GeneratePrimaryVertex(anEvent);
     }//old muline format
   }//useMuLineEvt
+  
+  else if(useAmBeEvt){ // Diego Costas (diego.costas.rodriguez@usc.es) 2023
+    // Create the energy distribution object
+    G4SPSEneDistribution* nEnergyDist = new G4SPSEneDistribution();
+    G4SPSRandomGenerator* rGen = new G4SPSRandomGenerator();
```

Comment:
Memory leaks here - `new` called on every event, never `delete`d. 
Please do the same as e.g. `LIGen`

Reply:
You should also be able to do some things on the first event (rather than every event) too like this (caching some things by storing them as private class data members)
```
   nEnergyDist->SetEnergyDisType("Arb");
    nEnergyDist->ArbEnergyHistoFile("../data/resampled_nSpectrum.txt");

    G4ParticleTable* particleTable = G4ParticleTable::GetParticleTable();
    G4ParticleDefinition* neutronDefinition = particleTable->FindParticle("neutron");
```